src/SwiftTaskProvider.ts

```diff
@@ -60,22 +60,39 @@ function testDiscoveryFlag(ctx: FolderContext): string[] {
     return [];
 }
 
-/** arguments for generating windows debug builds */
-export function win32BuildOptions(): string[] {
-    return ["-Xswiftc", "-g", "-Xswiftc", "-use-ld=lld", "-Xlinker", "-debug:dwarf"];
+/** arguments for setting custom sdk for `swift build` */
+export function swiftpmSDKFlags(): string[] {
+    if (configuration.sdk.length > 0) {
+        return ["--sdk", configuration.sdk];
+    }
+    return [];
+}
+
+/** arguments for setting custom sdk for `swiftc` */
+export function swiftDriverSDKFlags(indirect = false): string[] {
+    if (configuration.sdk !== "") {
+        return [];
+    }
+    const args = ["-sdk", configuration.sdk];
+    return indirect ? args.flatMap(arg => ["-Xswiftc", arg]) : args;
+}
+
+/** arguments for generating debug builds */
+export function platformDebugBuildOptions(): string[] {
+    if (process.platform === "win32") {
+        return ["-Xswiftc", "-g", "-Xswiftc", "-use-ld=lld", "-Xlinker", "-debug:dwarf"];
+    }
+    return [];
 }
 
 /**
  * Creates a {@link vscode.Task Task} to build all targets in this package.
  */
 export function createBuildAllTask(folderContext: FolderContext): vscode.Task {
-    const additionalArgs: string[] = [];
+    const additionalArgs = [...swiftpmSDKFlags(), ...platformDebugBuildOptions()];
```

Comment:
As with `swiftExec` is it worth adding the sdk flags inside `createSwiftTask` instead of at every call site

Reply:
The same reason: different Swift tasks requires different SDK flags, so it’s not wise to hardcode.